script: Chain up `scrollIntoView()` scrolling to parent `<iframe>`s

servo-wpt-sync opened the pull request:
Calling `scrollIntoView()` on an element within an `<iframe>` will now scroll scrolling boxes from the parent document(s), as long as they have the same origin.

Testing: One existing subtest passes, and adding a new test.

Reviewed in servo/servo#39475

Loirooriol commented:
@gsnedders Can you help me understand why https://github.com/web-platform-tests/wpt/pull/55075/checks?check_run_id=51346665255 is failing?

```
[taskcluster 2025-09-26 16:05:55.078Z] Task ID: eR8lCRerRV2rTTrxMuHOnw
[taskcluster 2025-09-26 16:05:55.078Z] Worker ID: 5740023176243119945
[taskcluster 2025-09-26 16:05:55.078Z] Worker Group: us-central1-b
[taskcluster 2025-09-26 16:05:55.078Z] Worker Node Type: projects/757942385826/machineTypes/n2-standard-4
[taskcluster 2025-09-26 16:05:55.078Z] Worker Pool: proj-wpt/ci
[taskcluster 2025-09-26 16:05:55.078Z] Worker Version: 46.1.0
[taskcluster 2025-09-26 16:05:55.078Z] Public IP: 34.123.48.91
[taskcluster 2025-09-26 16:05:55.078Z] Hostname: proj-wpt-ci-i3aeq2zxsz2anxbef0zbbw

8f84a9f2102e - Started downloading
40516d781e70 - Started downloading
ac811f28856d - Started downloading
ac811f28856d - Downloaded in 0.035 seconds
8f84a9f2102e - Downloaded in 0.204 seconds
44ebbf800033 - Started downloading
44ebbf800033 - Downloaded in 0.044 seconds
a8884b21e40e - Started downloading
4f4fb700ef54 - Started downloading
4f4fb700ef54 - Downloaded in 0.001 seconds
a8884b21e40e - Downloaded in 0.383 seconds

...(399 lines hidden)...

  wpt-servo-nightly-testharness-15
  wpt-servo-nightly-testharness-16
  wpt-servo-nightly-testharness-2
  wpt-servo-nightly-testharness-3
  wpt-servo-nightly-testharness-4
  wpt-servo-nightly-testharness-5
  wpt-servo-nightly-testharness-6
  wpt-servo-nightly-testharness-7
  wpt-servo-nightly-testharness-8
  wpt-servo-nightly-testharness-9
  wpt-servo-nightly-wdspec-1
  wpt-servo-nightly-wdspec-2
  wpt-webkitgtk_minibrowser-beta-crashtest-1
  wpt-webkitgtk_minibrowser-beta-reftest-1
  wpt-webkitgtk_minibrowser-beta-reftest-2
  wpt-webkitgtk_minibrowser-beta-reftest-3
  wpt-webkitgtk_minibrowser-beta-reftest-4
  wpt-webkitgtk_minibrowser-beta-reftest-5
  wpt-webkitgtk_minibrowser-beta-reftest-6
  wpt-webkitgtk_minibrowser-beta-testharness-1
  wpt-webkitgtk_minibrowser-beta-testharness-10
  wpt-webkitgtk_minibrowser-beta-testharness-11
  wpt-webkitgtk_minibrowser-beta-testharness-12
  wpt-webkitgtk_minibrowser-beta-testharness-13
  wpt-webkitgtk_minibrowser-beta-testharness-14
  wpt-webkitgtk_minibrowser-beta-testharness-15
  wpt-webkitgtk_minibrowser-beta-testharness-16
  wpt-webkitgtk_minibrowser-beta-testharness-2
  wpt-webkitgtk_minibrowser-beta-testharness-3
  wpt-webkitgtk_minibrowser-beta-testharness-4
  wpt-webkitgtk_minibrowser-beta-testharness-5
  wpt-webkitgtk_minibrowser-beta-testharness-6
  wpt-webkitgtk_minibrowser-beta-testharness-7
  wpt-webkitgtk_minibrowser-beta-testharness-8
  wpt-webkitgtk_minibrowser-beta-testharness-9
  wpt-webkitgtk_minibrowser-beta-wdspec-1
  wpt-webkitgtk_minibrowser-beta-wdspec-2
  wpt-webkitgtk_minibrowser-nightly-crashtest-1
  wpt-webkitgtk_minibrowser-nightly-reftest-1
  wpt-webkitgtk_minibrowser-nightly-reftest-2
  wpt-webkitgtk_minibrowser-nightly-reftest-3
  wpt-webkitgtk_minibrowser-nightly-reftest-4
  wpt-webkitgtk_minibrowser-nightly-reftest-5
  wpt-webkitgtk_minibrowser-nightly-reftest-6
  wpt-webkitgtk_minibrowser-nightly-testharness-1
  wpt-webkitgtk_minibrowser-nightly-testharness-10
  wpt-webkitgtk_minibrowser-nightly-testharness-11
  wpt-webkitgtk_minibrowser-nightly-testharness-12
  wpt-webkitgtk_minibrowser-nightly-testharness-13
  wpt-webkitgtk_minibrowser-nightly-testharness-14
  wpt-webkitgtk_minibrowser-nightly-testharness-15
  wpt-webkitgtk_minibrowser-nightly-testharness-16
  wpt-webkitgtk_minibrowser-nightly-testharness-2
  wpt-webkitgtk_minibrowser-nightly-testharness-3
  wpt-webkitgtk_minibrowser-nightly-testharness-4
  wpt-webkitgtk_minibrowser-nightly-testharness-5
  wpt-webkitgtk_minibrowser-nightly-testharness-6
  wpt-webkitgtk_minibrowser-nightly-testharness-7
  wpt-webkitgtk_minibrowser-nightly-testharness-8
  wpt-webkitgtk_minibrowser-nightly-testharness-9
  wpt-webkitgtk_minibrowser-nightly-wdspec-1
  wpt-webkitgtk_minibrowser-nightly-wdspec-2
  wpt-webkitgtk_minibrowser-stable-crashtest-1
  wpt-webkitgtk_minibrowser-stable-reftest-1
  wpt-webkitgtk_minibrowser-stable-reftest-2
  wpt-webkitgtk_minibrowser-stable-reftest-3
  wpt-webkitgtk_minibrowser-stable-reftest-4
  wpt-webkitgtk_minibrowser-stable-reftest-5
  wpt-webkitgtk_minibrowser-stable-reftest-6
  wpt-webkitgtk_minibrowser-stable-testharness-1
  wpt-webkitgtk_minibrowser-stable-testharness-10
  wpt-webkitgtk_minibrowser-stable-testharness-11
  wpt-webkitgtk_minibrowser-stable-testharness-12
  wpt-webkitgtk_minibrowser-stable-testharness-13
  wpt-webkitgtk_minibrowser-stable-testharness-14
  wpt-webkitgtk_minibrowser-stable-testharness-15
  wpt-webkitgtk_minibrowser-stable-testharness-16
  wpt-webkitgtk_minibrowser-stable-testharness-2
  wpt-webkitgtk_minibrowser-stable-testharness-3
  wpt-webkitgtk_minibrowser-stable-testharness-4
  wpt-webkitgtk_minibrowser-stable-testharness-5
  wpt-webkitgtk_minibrowser-stable-testharness-6
  wpt-webkitgtk_minibrowser-stable-testharness-7
  wpt-webkitgtk_minibrowser-stable-testharness-8
  wpt-webkitgtk_minibrowser-stable-testharness-9
  wpt-webkitgtk_minibrowser-stable-wdspec-1
  wpt-webkitgtk_minibrowser-stable-wdspec-2
  wpt-wpewebkit_minibrowser-beta-crashtest-1
  wpt-wpewebkit_minibrowser-beta-reftest-1
  wpt-wpewebkit_minibrowser-beta-reftest-2
  wpt-wpewebkit_minibrowser-beta-reftest-3
  wpt-wpewebkit_minibrowser-beta-reftest-4
  wpt-wpewebkit_minibrowser-beta-reftest-5
  wpt-wpewebkit_minibrowser-beta-reftest-6
  wpt-wpewebkit_minibrowser-beta-testharness-1
  wpt-wpewebkit_minibrowser-beta-testharness-10
  wpt-wpewebkit_minibrowser-beta-testharness-11
  wpt-wpewebkit_minibrowser-beta-testharness-12
  wpt-wpewebkit_minibrowser-beta-testharness-13
  wpt-wpewebkit_minibrowser-beta-testharness-14
  wpt-wpewebkit_minibrowser-beta-testharness-15
  wpt-wpewebkit_minibrowser-beta-testharness-16
  wpt-wpewebkit_minibrowser-beta-testharness-2
  wpt-wpewebkit_minibrowser-beta-testharness-3
  wpt-wpewebkit_minibrowser-beta-testharness-4
  wpt-wpewebkit_minibrowser-beta-testharness-5
  wpt-wpewebkit_minibrowser-beta-testharness-6
  wpt-wpewebkit_minibrowser-beta-testharness-7
  wpt-wpewebkit_minibrowser-beta-testharness-8
  wpt-wpewebkit_minibrowser-beta-testharness-9
  wpt-wpewebkit_minibrowser-beta-wdspec-1
  wpt-wpewebkit_minibrowser-beta-wdspec-2
  wpt-wpewebkit_minibrowser-nightly-crashtest-1
  wpt-wpewebkit_minibrowser-nightly-reftest-1
  wpt-wpewebkit_minibrowser-nightly-reftest-2
  wpt-wpewebkit_minibrowser-nightly-reftest-3
  wpt-wpewebkit_minibrowser-nightly-reftest-4
  wpt-wpewebkit_minibrowser-nightly-reftest-5
  wpt-wpewebkit_minibrowser-nightly-reftest-6
  wpt-wpewebkit_minibrowser-nightly-testharness-1
  wpt-wpewebkit_minibrowser-nightly-testharness-10
  wpt-wpewebkit_minibrowser-nightly-testharness-11
  wpt-wpewebkit_minibrowser-nightly-testharness-12
  wpt-wpewebkit_minibrowser-nightly-testharness-13
  wpt-wpewebkit_minibrowser-nightly-testharness-14
  wpt-wpewebkit_minibrowser-nightly-testharness-15
  wpt-wpewebkit_minibrowser-nightly-testharness-16
  wpt-wpewebkit_minibrowser-nightly-testharness-2
  wpt-wpewebkit_minibrowser-nightly-testharness-3
  wpt-wpewebkit_minibrowser-nightly-testharness-4
  wpt-wpewebkit_minibrowser-nightly-testharness-5
  wpt-wpewebkit_minibrowser-nightly-testharness-6
  wpt-wpewebkit_minibrowser-nightly-testharness-7
  wpt-wpewebkit_minibrowser-nightly-testharness-8
  wpt-wpewebkit_minibrowser-nightly-testharness-9
  wpt-wpewebkit_minibrowser-nightly-wdspec-1
  wpt-wpewebkit_minibrowser-nightly-wdspec-2
  wpt-wpewebkit_minibrowser-stable-crashtest-1
  wpt-wpewebkit_minibrowser-stable-reftest-1
  wpt-wpewebkit_minibrowser-stable-reftest-2
  wpt-wpewebkit_minibrowser-stable-reftest-3
  wpt-wpewebkit_minibrowser-stable-reftest-4
  wpt-wpewebkit_minibrowser-stable-reftest-5
  wpt-wpewebkit_minibrowser-stable-reftest-6
  wpt-wpewebkit_minibrowser-stable-testharness-1
  wpt-wpewebkit_minibrowser-stable-testharness-10
  wpt-wpewebkit_minibrowser-stable-testharness-11
  wpt-wpewebkit_minibrowser-stable-testharness-12
  wpt-wpewebkit_minibrowser-stable-testharness-13
  wpt-wpewebkit_minibrowser-stable-testharness-14
  wpt-wpewebkit_minibrowser-stable-testharness-15
  wpt-wpewebkit_minibrowser-stable-testharness-16
  wpt-wpewebkit_minibrowser-stable-testharness-2
  wpt-wpewebkit_minibrowser-stable-testharness-3
  wpt-wpewebkit_minibrowser-stable-testharness-4
  wpt-wpewebkit_minibrowser-stable-testharness-5
  wpt-wpewebkit_minibrowser-stable-testharness-6
  wpt-wpewebkit_minibrowser-stable-testharness-7
  wpt-wpewebkit_minibrowser-stable-testharness-8
  wpt-wpewebkit_minibrowser-stable-testharness-9
  wpt-wpewebkit_minibrowser-stable-wdspec-1
  wpt-wpewebkit_minibrowser-stable-wdspec-2
INFO:root:SCHEDULED TASKS:
  lint
  sink-task
  wpt-chrome-dev-results
  wpt-chrome-dev-results-without-changes
  wpt-chrome-dev-stability
  wpt-firefox-nightly-results
  wpt-firefox-nightly-results-without-changes
  wpt-firefox-nightly-stability
INFO:root:Scheduling sink-task
Traceback (most recent call last):
  File "/home/test/web-platform-tests/_venv3/lib/python3.10/site-packages/taskcluster/client.py", line 503, in _makeHttpRequest
    response.raise_for_status()
  File "/home/test/web-platform-tests/_venv3/lib/python3.10/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 406 Client Error: Not Acceptable for url: http://taskcluster/api/queue/v1/task/GVZ6FL2kQ3iVIdrWt-W7Bw

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/test/web-platform-tests/./wpt", line 10, in <module>
    wpt.main()
  File "/home/test/web-platform-tests/tools/wpt/wpt.py", line 233, in main
    rv = script(*args, **kwargs)
  File "/home/test/web-platform-tests/tools/ci/tc/decision.py", line 408, in run
    create_tasks(queue, task_id_map)
  File "/home/test/web-platform-tests/tools/ci/tc/decision.py", line 349, in create_tasks
    queue.createTask(task_id, task_data)
  File "/home/test/web-platform-tests/_venv3/lib/python3.10/site-packages/taskcluster/generated/queue.py", line 296, in createTask
    return self._makeApiCall(self.funcinfo["createTask"], *args, **kwargs)
  File "/home/test/web-platform-tests/_venv3/lib/python3.10/site-packages/taskcluster/client.py", line 269, in _makeApiCall
    response = self._makeHttpRequest(entry['method'], _route, payload)
  File "/home/test/web-platform-tests/_venv3/lib/python3.10/site-packages/taskcluster/client.py", line 538, in _makeHttpRequest
    raise exceptions.TaskclusterRestFailure(
taskcluster.exceptions.TaskclusterRestFailure: None
[taskcluster 2025-09-26 16:07:39.236Z] === Task Finished ===
[taskcluster 2025-09-26 16:07:39.661Z] Unsuccessful task run with exit code: 1 completed in 104.586 seconds
```

Loirooriol committed:
script: Chain up `scrollIntoView()` scrolling to parent `<iframe>`s

Calling `scrollIntoView()` on an element within an `<iframe>` will now
scroll scrolling boxes from the parent document(s), as long as they have
the same origin.

Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: Oriol Brufau <obrufau@igalia.com>

servo-wpt-sync commented:
⛔ The downstream PR has merged (servo/servo#39475), but these changes could not be merged properly. Please address any CI issues and try to merge manually.

Loirooriol commented:
Landed in #55119
